File: MegaHerdt.Models/Models/ReparationClaim.cs
using MegaHerdt.Models.Models.Identity;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MegaHerdt.Models.Models
{
    public class ReparationClaim
    {
        [Key]
        public int Id { get; set; }
        [ForeignKey("User")]
        public string ClientId { get; set; }
        [ForeignKey("Reparation")]
        public int ReparationId { get; set; }
        public string Description { get; set; }
        public DateTime Date { get; set; }
        public bool Answered { get; set; } = false;
        public User Client { get; set; }
        public Reparation Reparation { get; set; }
    }
}

File: MegaHerdt.FrontEnd/MegaHerdt/src/app/models/ReparationClaims/ReparationClaim.ts
import { ReparationDetail } from "../Reparation/ReparationDetail";

export class ReparationClaim{
    id: number;
    description: string;
    date: Date;
    answered: boolean;
    reparation: ReparationDetail;
    constructor(id: number, description: string, date: Date, answered: boolean, reparation: ReparationDetail){
        this.id = id;
        this.description = description;
        this.date = date;
        this.answered = answered;
        this.reparation = reparation;
    }
}